pkg/extensions/search/schema.resolvers.go

```diff
@@ -463,6 +463,74 @@ func (r *queryResolver) GlobalSearch(ctx context.Context, query string) (*gql_ge
 	}, nil
 }
 
+// DependencyListForImage is the resolver for the DependencyListForImage field.
+func (r *queryResolver) DerivedImageList(ctx context.Context, image string) ([]*gql_generated.ImageSummary, error) {
```

Comment:
We need to make sure all of the layers in the image sent as parameter are layers in the images returned by this method.
We don't what to know if the images have the exact same layers.
We want images with layers which are a superset of the set of layers sent in the image sent as parameter.

So if image `A` given as parameter has 3 layers `a b c`, and  image `B` has 4 layers `a b c d`, B meets the condition to be derived from A.
So if image `A` given as parameter has 3 layers `a b c`, and  image `C` has 2 layers `a b`, B meets the condition to be a base image for A (good for your other PR for BaseImageList), but doesn't meet the condition for this query.

Reply:
I added tests to verify all possible cases, equal layers, less layers than A and more layers than A.